js/main.js

```diff
@@ -38,14 +38,15 @@ $('.reservation-button').on('click', function(event) {
   reservationReference.push(reservationData);
 });
 // Create reservation function
-//function getReservations() {
-  // listen any changes to the database 
+function getReservations() {
+  // listen any changes to the database
   database.ref('reservations').on('value', function(snapshot) {
     // get all reservations in the results from the database
     //var reservationList = $('.reservation-list');
     var allReservations = snapshot.val();
     // remove any reservations the are that displaied current reservations list
-    // iterate through each reservation from database 
+    $('.reservation-list').empty();
```

Comment:
How did you know to put this empty action line here? I want to understand the logic. I thought the list should be empty right before append the new reservation to the list.

Reply:
You are correct that it should be empty before appending a new reservation to the list, but where you had it before it would empty before you appended EACH new reservation to the list. Effectively, you would add a reservation to the list, then empty the list, then add the next reservation, then empty it, and so on. This is why you ended up just seeing one on the list.

This way, you empty the list before ANY new reservations are added, which means that the list can re-populate with all the updated values.